lib/auth/clusterconfig/clusterconfigv1/service_test.go

```diff
@@ -1755,3 +1764,111 @@ func newTestEnv(opts ...serviceOpt) (*env, error) {
 		emitter:                 emitter,
 	}, nil
 }
+
+func TestGetAccessGraphConfig(t *testing.T) {
+	cfgEnabled := clusterconfigv1.AccessGraphConfig{
+		Enabled:  true,
+		Address:  "address",
+		CA:       []byte("ca"),
+		Insecure: true,
+	}
+	cases := []struct {
+		name              string
+		accessGraphConfig clusterconfigv1.AccessGraphConfig
+		role              types.SystemRole
+		testSetup         func(*testing.T)
+		errorAssertion    require.ErrorAssertionFunc
+		responseAssertion *clusterconfigpb.GetClusterAccessGraphConfigResponse
+	}{
+		{
+			name: "unauthorized",
+			role: types.RoleKube,
+			errorAssertion: func(t require.TestingT, err error, _ ...any) {
+				require.True(t, trace.IsAccessDenied(err), "got (%v), expected unauthorized user to be prevented from getting auth preferences", err)
+			},
+		},
+		{
+			name:              "authorized proxy with non empty access graph config; Policy module is disabled",
+			role:              types.RoleProxy,
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled: false,
+				},
+			},
+		},
+		{
+			name: "authorized proxy with non empty access graph config; Policy module is enabled",
+			role: types.RoleProxy,
+			testSetup: func(t *testing.T) {
+				m := modules.TestModules{
+					TestFeatures: modules.Features{
+						Policy: modules.PolicyFeature{
+							Enabled: true,
+						},
+					},
+				}
+				modules.SetTestModules(t, &m)
+			},
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled:  true,
+					Insecure: true,
+					Address:  "address",
+					Ca:       []byte("ca"),
+				},
+			},
+		},
+		{
+			name: "authorized discovery with non empty access graph config; Policy module is enabled",
+			role: types.RoleDiscovery,
+			testSetup: func(t *testing.T) {
+				m := modules.TestModules{
+					TestFeatures: modules.Features{
+						Policy: modules.PolicyFeature{
+							Enabled: true,
+						},
+					},
+				}
+				modules.SetTestModules(t, &m)
+			},
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled:  true,
+					Insecure: true,
+					Address:  "address",
+					Ca:       []byte("ca"),
+				},
+			},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.testSetup != nil {
```

Comment:
nit: I'd remove this check and let the test panic if the setup function is not set. Panic here is less evil than forgetting to set the setup function, even if it's empty.